plugins/nf-nomad/src/main/nextflow/nomad/executor/NomadService.groovy

```diff
@@ -300,12 +316,21 @@ class NomadService implements Closeable{
 
     protected void purgeJob(String jobId, boolean purge){
         log.debug "[NOMAD] purgeJob with jobId=${jobId}"
-        jobsApi.deleteJob(jobId,config.jobOpts().region, config.jobOpts().namespace,null,null,purge, true)
+        try {
+            jobsApi.deleteJob(jobId, config.jobOpts().region, config.jobOpts().namespace, null, null, purge, true)
+        }catch(Exception e){
+            log.error("[NOMAD] Failed to delete job ${jobId} -- Cause: ${e.message ?: e}", e)
```

Comment:
@jagedn , do you think that within an exception block, we should use only `log.debug` or this should be decided as per the context?

I think `error` makes sense for session terminating steps and not for the retries - what do you think?

Reply:
good question

yes, you're right, probably we can use only debug level in retries and if the user want to have more details can change the level

Reply:
Thanks Jorge, I've refactored accordingly 👍